File: src/api/audit.ts
import { call } from './index'

export interface AuditEntry {
  id: string
  actor_id: string | null
  actor_name: string | null
  action: string
  resource: string | null
  target_id: string | null
  payload: string | null
  ip: string | null
  created_at: string
}

export interface AuditListResult {
  items: AuditEntry[]
  total: number
  page: number
  page_size: number
}

export interface AuditFilter {
  action?: string
  actor_id?: string
  resource?: string
  payload_search?: string
  from?: string
  to?: string
}

export const auditApi = {
  list: (token: string, filter: AuditFilter = {}, page = 1, pageSize = 50) =>
    call<AuditListResult>('audit_list', {
      token,
      query: {
        action: filter.action || null,
        actor_id: filter.actor_id || null,
        resource: filter.resource || null,
        payload_search: filter.payload_search || null,
        from: filter.from || null,
        to: filter.to || null,
        page,
        page_size: pageSize
      }
    })
}

File: src-tauri/src/models/audit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AuditLogEntry {
    pub id: String,
    pub actor_id: Option<String>,
    pub actor_name: Option<String>,
    pub action: String,
    pub resource: Option<String>,
    pub target_id: Option<String>,
    pub payload: Option<String>,
    pub ip: Option<String>,
    pub created_at: String,
}

/// All filter fields are optional.  `from` / `to` are ISO-8601 strings that
/// get compared against `created_at` lexically — that works because we write
/// timestamps in the canonical `YYYY-MM-DDTHH:MM:SS.fffZ` format.  If the
/// frontend wants "last 24h" it can compute `now - 86400` itself.
#[derive(Debug, Clone, Deserialize)]
pub struct AuditQuery {
    pub action: Option<String>,
    pub actor_id: Option<String>,
    pub resource: Option<String>,
    pub payload_search: Option<String>,
    pub from: Option<String>,
    pub to: Option<String>,
    pub page: Option<i64>,
    pub page_size: Option<i64>,
}